distribution/pull_v2.go

```diff
@@ -15,26 +15,23 @@ import (
 	"github.com/docker/distribution/manifest/schema1"
 	"github.com/docker/distribution/reference"
 	"github.com/docker/docker/distribution/metadata"
+	"github.com/docker/docker/distribution/xfer"
 	"github.com/docker/docker/image"
 	"github.com/docker/docker/image/v1"
 	"github.com/docker/docker/layer"
-	"github.com/docker/docker/pkg/archive"
-	"github.com/docker/docker/pkg/broadcaster"
-	"github.com/docker/docker/pkg/progressreader"
-	"github.com/docker/docker/pkg/streamformatter"
+	"github.com/docker/docker/pkg/ioutils"
 	"github.com/docker/docker/pkg/stringid"
 	"github.com/docker/docker/registry"
 	"golang.org/x/net/context"
 )
 
 type v2Puller struct {
+	ctx            context.Context
```

Comment:
> You should pass a context with the functions you call and make a new ones if you need to split to concurrent goroutines.

We definitely break this rule. Typically, we break the rule where all of the function calls will use the same context, such as with a reader or when doing serialization, since we can't change the method signature to include the context.

@aaronlehmann Are we doing this to avoid passing context around with puller calls?